dts/arm/nxp/nxp_imx.dtsi

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2017, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <dt-bindings/clock/imx_ccm.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4";
+			reg = <0>;
+		};
+	};
+
+	soc {
+
+		ddr_code: code@10000000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x10000000 0xfff0000>;
+			label = "DDR CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ddr_sys: memory@80000000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x80000000 0x60000000>;
+			label = "DDR SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		tcml_code: code@1fff8000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x1fff8000 0x8000>;
+			label = "TCML CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		tcmu_sys: memory@20000000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x20000000 0x8000>;
+			label = "TCMU SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_code: code@00900000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x00900000 0x20000>;
+			label = "OCRAM CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_sys: memory@20200000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x20200000 0x20000>;
+			label = "OCRAM SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_s_code: code@20180000 {
+			compatible = "nxp,imx-code-bus";
+			reg = <0x20180000 0x8000>;
+			label = "OCRAM_S CODE";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_s_sys: memory@00000000 {
+			device_type = "memory";
+			compatible = "nxp,imx-sys-bus";
+			reg = <0x00000000 0x8000>;
+			label = "OCRAM_S SYSTEM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		ocram_pxp_code: code@00940000 {
```

Comment:
Does it really make sense to have pxp & edpc used for memory/code storage for Zephyr?  It seems like TCM would be the default, and exposing OCRAM_S, OCRAM, and DDR would be more than enough.

Reply:
As soon as pxp and edpc can be used for this purpose I thought that should be a good idea to have them exposed as well.

Reply:
I'm not opposed, but assume that PXP & EDPC are meant to normally be used by the various graphics devices/drivers that they are associated with.  Also, I'd assume the performance of going across various SoC busses to those memory is going to be higher compared to TCM.  It just feels rare that PXP or EDPC memories would be what one would choose, but someone could tell me otherwise.